### PR TITLE
New slide set “Loops for Arrays"

### PR DESCRIPTION
**Pull request by lxxxvi**
Lesson 4 lacks information about how to loop/iterate through an array. There are exercises that require that know how. Here's a proposal how we could fill in this gap.

This slide set explains how we can use loops on arrays. In the end it introduces `Array#each`.

### CODE REVIEW

File: slides/loops-for-arrays/index.html [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    break                     # this exits the loops
+  end
+end
+</code></pre>
+          </div>
+        </section>
+
+        <section>
+          <h2>One more thing</h2>
+          <p>The abort condition <code>if counter < 3</code> needs improvment.</p>
```
Comment (nerdinand) [inline slides/loops-for-arrays/index.html:157]: typo: improvment

File: slides/loops-for-arrays/index.html [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+end
+</pre></code>
+          </div>
+        </section>
+
+        <section>
+          <h2><code>.each</code> Cont. II</h2>
+          <p class="fragment fade-in">It's a <code>loop</code> .</p>
+          <p class="fragment fade-in">There's no (visible) <code>counter</code> .</p>
+          <p class="fragment fade-in">There's no (visible) abortion criterea.</p>
```
[the remark this answers is not in the document]
Comment (lxxxvi) [reply slides/loops-for-arrays/index.html]: It took me a Google Search to see what the typo was... 🙈 😂

**Comment by nerdinand** [comment]
> I'd merge it now and later add an example for 'each' with numbers maybe?

Do you mean `.each.with_index`?

**Comment by hharen** [comment]
No, I meant an example for using each for an array of numbers? Just to show
them more cases where each is used. I think for each.with_index is a bit
too soon...

email: h.harencarova@gmail.com
website: seionline.ch
website: momslearntocode.com
mobile CH: +41 78 737 32 33
mobile SK: +421 915 579 068

**Comment by nerdinand** [comment]
Yeah, I think it would be good to have more examples eventually, we just need to think of something reasonable... I think that will also help to explain the `|variable|` with more examples, because that's always confusing (the fact that it's really just a regular old variable, but only visible inside the block).

**Comment by hharen** [comment]
yes, exactly that. but I'd merge it now, so that we could use it tomorrow.

email: h.harencarova@gmail.com
website: seionline.ch
website: momslearntocode.com
mobile CH: +41 78 737 32 33
mobile SK: +421 915 579 068

**Comment by nerdinand** [comment]
Agreed, @lxxxvi merge it! And I'll create a follow-up issue for more examples and explanation.

**Comment by nerdinand** [comment]
Closes #25

**Comment by lxxxvi** [comment]
@hharen @nerdinand I added two more examples

**Comment by nerdinand** [comment]
Beautiful, thanks @lxxxvi!

**Pull request merged**